Fix Sample C Source File Warning Error

This issue suggests fixing the following build error generated after testing the project on macOS using the default compiler:

```
Internal ctest changing into directory: /Users/threeal/Workspaces/cmake/CheckWarning.cmake/build
Test project /Users/threeal/Workspaces/cmake/CheckWarning.cmake/build
    Start 1: Check warning for success
1/6 Test #1: Check warning for success ........................***Failed    2.44 sec
-- Reconfiguring sample project
CMake Error at Assertion.cmake:123 (message):
  expected command:

    /usr/local/Cellar/cmake/3.29.0/bin/cmake --build /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/sample/build

  not to fail with error:

    /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/sample/src/main.c:3:9: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
    int main() {
            ^
             void
    1 error generated.
    make[2]: *** [CMakeFiles/main_c.dir/src/main.c.o] Error 1
    make[1]: *** [CMakeFiles/main_c.dir/all] Error 2
    make: *** [all] Error 2
    
Call Stack (most recent call first):
  Assertion.cmake:488 (fail)
  /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/CheckWarningTest.cmake:36 (assert_execute_process)
  /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/CheckWarningTest.cmake:43 (build_sample)
  /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/CheckWarningTest.cmake:71 (Check warning for success)
  /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/CheckWarningTest.cmake:71 (cmake_language)



    Start 2: Check warning globally for success
2/6 Test #2: Check warning globally for success ...............***Failed    1.42 sec
-- Reconfiguring sample project
CMake Error at Assertion.cmake:123 (message):
  expected command:

    /usr/local/Cellar/cmake/3.29.0/bin/cmake --build /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/sample/build

  not to fail with error:

    /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/sample/src/main.c:3:9: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
    int main() {
            ^
             void
    1 error generated.
    make[2]: *** [CMakeFiles/main_c.dir/src/main.c.o] Error 1
    make[1]: *** [CMakeFiles/main_c.dir/all] Error 2
    make: *** [all] Error 2
    
Call Stack (most recent call first):
  Assertion.cmake:488 (fail)
  /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/CheckWarningTest.cmake:36 (assert_execute_process)
  /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/CheckWarningTest.cmake:48 (build_sample)
  /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/CheckWarningTest.cmake:71 (Check warning globally for success)
  /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/CheckWarningTest.cmake:71 (cmake_language)



    Start 3: Check warning for failure
3/6 Test #3: Check warning for failure ........................   Passed    1.16 sec
    Start 4: Check warning globally for failure
4/6 Test #4: Check warning globally for failure ...............   Passed    1.13 sec
    Start 5: Check warning for failure but ignored
5/6 Test #5: Check warning for failure but ignored ............***Failed    1.38 sec
-- Reconfiguring sample project
CMake Error at Assertion.cmake:123 (message):
  expected command:

    /usr/local/Cellar/cmake/3.29.0/bin/cmake --build /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/sample/build

  not to fail with error:

    /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/sample/src/main.c:3:9: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
    int main() {
            ^
             void
    1 error generated.
    make[2]: *** [CMakeFiles/main_c.dir/src/main.c.o] Error 1
    make[1]: *** [CMakeFiles/main_c.dir/all] Error 2
    make: *** [all] Error 2
    
Call Stack (most recent call first):
  Assertion.cmake:488 (fail)
  /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/CheckWarningTest.cmake:36 (assert_execute_process)
  /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/CheckWarningTest.cmake:63 (build_sample)
  /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/CheckWarningTest.cmake:71 (Check warning for failure but ignored)
  /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/CheckWarningTest.cmake:71 (cmake_language)



    Start 6: Check warning globally for failure but ignored
6/6 Test #6: Check warning globally for failure but ignored ...***Failed    1.41 sec
-- Reconfiguring sample project
CMake Error at Assertion.cmake:123 (message):
  expected command:

    /usr/local/Cellar/cmake/3.29.0/bin/cmake --build /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/sample/build

  not to fail with error:

    /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/sample/src/main.c:3:9: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
    int main() {
            ^
             void
    1 error generated.
    make[2]: *** [CMakeFiles/main_c.dir/src/main.c.o] Error 1
    make[1]: *** [CMakeFiles/main_c.dir/all] Error 2
    make: *** [all] Error 2
    
Call Stack (most recent call first):
  Assertion.cmake:488 (fail)
  /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/CheckWarningTest.cmake:36 (assert_execute_process)
  /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/CheckWarningTest.cmake:68 (build_sample)
  /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/CheckWarningTest.cmake:71 (Check warning globally for failure but ignored)
  /Users/threeal/Workspaces/cmake/CheckWarning.cmake/test/CheckWarningTest.cmake:71 (cmake_language)




33% tests passed, 4 tests failed out of 6

Total Test time (real) =   8.95 sec

The following tests FAILED:
          1 - Check warning for success (Failed)
          2 - Check warning globally for success (Failed)
          5 - Check warning for failure but ignored (Failed)
          6 - Check warning globally for failure but ignored (Failed)
Errors while running CTest
```

The error is likely caused by a missing `void` parameter in the `main` function of the `main.c` file.

--- a/test/sample/src/main.c
+++ b/test/sample/src/main.c
@@ -1,6 +1,6 @@
 #include <success.h>
 
-int main() {
+int main(void) {
 #ifdef WITH_UNUSED
   int unused;
 #endif

